packages/grafana-data/src/transformations/fieldReducer.ts

```diff
@@ -553,3 +612,24 @@ function calculateDistinctCount(field: Field, ignoreNulls: boolean, nullAsZero:
   }
   return { distinctCount: distinct.size };
 }
+
+function calculatePercentile(
```

Comment:
should this be plural maybe, since we're already doing all the work to calculate one. shoul dbe cheap to calculate multiple percentiles. :thinking:

Reply:
I'm not sure how that would work in the way that reducers are used, but open to the idea. I did change the approach slightly to just be all the percentiles 1-99 instead of a select few based on convo with @baldm0mma

Reply:
hmm, yeah, maybe adding a cache somewhere in between (later) will work also.

how about not passing `name` here and just returning `sorted[index]`. then move the wrapper up to the `reduce` function above?

```js
    reduce: (field: Field, ignoreNulls: boolean, nullAsZero: boolean) => ({
      [id]: calculatePercentile(id, percentile, field, ignoreNulls, nullAsZero)
    })
```

Reply:
i would maybe swap the `percentile` and `field` args, also `calculatePercentile(field, percentile, ignoreNulls, nullAsZero)`

Reply:
That should be an easy change, will do

Reply:
Fixed in https://github.com/grafana/grafana/pull/85554/commits/5ca5ee22534ff9912cb90bb991b2b63ab4aa72f8

Reply:
ty